cmd/commands/namespace.go

```diff
@@ -49,7 +49,8 @@ func NamespaceInit(manifests map[string]*core.Manifest, kc *core.KubectlClient)
 		),
 		PreRunE: FlagsValidatorAsCobraRunE(
 			FlagsValidationConjunction(
-				AtMostOneOf("gcr", "dockerhub"),
+				AtMostOneOf("gcr", "dockerhub", "no-secret"),
+				AtMostOneOf("secret", "no-secret"),
 				NotBlank("secret"),
```

Comment:
not related to this PR necessarily, but does this `NotBlank` constraint mean the default specified in the help for "secret" is incorrect?

Reply:
the default is "push-credentials" so it is NotBlank

Reply:
good point :) so how would it ever be blank, if a literal `""` were provided?